.github/workflows/prettier.yaml

```diff
@@ -14,23 +17,38 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
         with:
-          ref: ${{github.event.pull_request.head.ref}}
-          repository: ${{github.event.pull_request.head.repo.full_name}}
-          fetch-depth: 0
-
-      - name: Check if PR is from a fork
-        id: check_fork
-        run: echo "::set-output name=is_fork::${{ github.event.pull_request.head.repo.fork }}"
+          fetch-depth: 0  # Needed to compare with origin/main
 
-      - name: Prettify code
-        id: prettify
-        uses: creyD/prettier_action@v4.6
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
         with:
-          prettier_options: --write **/index.html --print-width 200
-          only_changed: True
-          commit_message: "chore: prettier"
-        continue-on-error: true
-
-      - name: Handle commit failure
-        if: ${{ steps.prettify.outcome == 'failure' && steps.check_fork.outputs.is_fork == 'true' }}
-        run: echo "Please run prettier locally"
+          node-version: 'lts/*'
+
+      - name: Install Prettier
+        run: npm install --global prettier
+
+      - name: Get changed files and run Prettier
+        id: run_prettier
+        run: |
+          git fetch origin main
+
+          changed_files=$(git diff --name-only origin/main...HEAD | grep -E '\.(js|json|css|html)$' || true)
+
+          if [[ -n "$changed_files" ]]; then
+            echo "$changed_files" | xargs prettier --write --print-width 200
+            echo "changed=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "changed=false" >> "$GITHUB_OUTPUT"
+          fi
```

Comment:
I didn't know that kind of scripting worked. I didn't test it but it looks good.

Reply:
Tested on [my fork ](https://github.com/daniel-montalvo/aria/pull/5/commits)and I would say it does  do what we are expecting